railties/test/generators/shared_generator_tests.rb

```diff
@@ -362,7 +362,7 @@ def test_generator_when_sprockets_is_not_used
 
   def test_dev_option
     run_generator_using_prerelease [destination_root, "--dev"]
-    rails_path = File.expand_path("../../..", Rails.root)
+    rails_path = File.expand_path("../../..", __dir__)
     assert_file "Gemfile", %r{^gem ["']rails["'], path: ["']#{Regexp.escape rails_path}["']$}
```

Comment:
```suggestion
    assert_file "Gemfile", %r{^gem ["']rails["'], path: ["']#{Regexp.escape RAILS_DEV_PATH}["']$}
```

Reply:
Not available here, and I question whether it is worth requiring generators_test_helpers here. Already pretty clear that it is the rails dev path because of the gemfile path check.

Reply:
Ok no problem. By the way I don't seem to have the right to resolve comments, so if you could resolve this comment and the other outdated one that would result in an easier to read PR, thanks

Reply:
By the way, the constant would be available if it were defined above `module GeneratorsTestHelper` instead of inside it. Similar to how `RAILS_FRAMEWORK_ROOT` is defined.

Reply:
OH... how about we just define it in `railties/test/abstract_unit` and follow the same pattern as the isolation tests! Using the same name! That works for all the tests. 

Seems so obvious now 🤣 Thanks for the help!

Reply:
Nice! Tests are failing right now though with a constant already initialized error

Reply:
Yup! Turns out using the same constant name, in the global namespace, was not as brilliant as I thought!

I decided to use DEV_RAILS_PATH so it is noticeably different than the constant being tested, and leaving it in the GeneratorsTestHelper module, and not changing the shared test file where helpers aren't loaded.  

Could use the current global RAILS_FRAMEWORK_ROOT, but depends on require chains that I don't want to worry about!